lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -144,6 +144,28 @@ def visit_m_assign(node)
         end
       end
 
+      sig { params(node: SyntaxTree::Params).void }
+      def visit_params(node)
+        node.keywords.each do |keyword,|
+          location = keyword.location
+
+          # We exclude the ":" symbol from the keyword
+          # for consistency with the rest of the parameters
+          modified_location = SyntaxTree::Location.new(
+            start_line: location.start_line,
+            start_column: location.start_column,
+            start_char: location.start_char,
+            end_char: location.end_char - 1,
+            end_column: location.end_column - 1,
+            end_line: location.end_line,
+          )
```

Comment:
Should this be extracted as a private method?

The name of the extracted method would allow for self-documentation at the call-site, the current comment can be moved as part of the method documentation and the method could be reused for symbols as hash keys and for keyword arguments in call-sites if they have the same issue.

Reply:
https://github.com/Shopify/ruby-lsp/pull/142/commits/666abf163fbc6b6ee71c29566e75911272a0dd0b